main.py

```diff
@@ -1039,32 +1041,33 @@ def apidoc():
 def news():
     """ Handler for a page with a top news list """
     topic = request.args.get("topic")
-    start = request.args.get("start")
-    if start is not None:
-        try:
-            if "." in start:
-                # Assume full timestamp with microseconds
-                start = datetime.strptime(start, "%Y-%m-%dT%H:%M:%S.%f")
-            else:
-                # Compact timestamp
-                start = datetime.strptime(start, "%Y-%m-%dT%H:%M:%S")
-        except ValueError:
-            start = None
-    articles = top_news(topic=topic, start=start)
-    now = datetime.utcnow()
+    try:
+        offset = max(0, int(request.args.get("offset", 0)))
+        limit = max(0, int(request.args.get("limit", _TOP_NEWS_LENGTH)))
+    except:
+        offset = 0
+        limit = _TOP_NEWS_LENGTH
+
+    articles = top_news(topic=topic, offset=offset)
```

Comment:
Clamp `limit` to a maximum value here (avoiding cases where people manually put in 100 billion or whatever)?

Reply:
...and the same applies to `offset` also, btw.

Reply:
Well spotted wrt limit. Offset could be 100 billion without trouble, would just return 0 results.